### PR TITLE
Added gz harmonic + ros humble as multistage build

### PR DESCRIPTION
**Pull request by fzoric8**
* Added multistage build for the gz harmonic, you can build gazebo with adding `target` argument to the `docker build` command as: 
```
docker build -t ros2_img:humble --target ign_gazebo
```
* Added `chmod +x` for `run_docker.sh` 
* Added `developer` user to put workspace into `/home/developer`
* changed `display=$display` to `DISPLAY=$DISPLAY` to enable GUI for GZ

### CODE REVIEW

File: ros2-humble/custom/Dockerfile [hunk 1]
```diff
@@ -62,15 +60,49 @@ COPY ./to_copy/nanorc /root/.nanorc
...
+    ros-${ROS2_DISTRO}-ros2-control \
+    ros-${ROS2_DISTRO}-ros2-controllers \
+    ros-${ROS2_DISTRO}-hardware-interface \
+    ros-${ROS2_DISTRO}-controller-interface \ 
+    ros-${ROS2_DISTRO}-controller-manager \
+    libignition-transport11-dev \
+    libignition-gazebo6-dev 
+
+WORKDIR /home/developer/ros2_ws/src
+RUN git clone https://github.com/gazebosim/ros_gz.git -b ${ROS2_DISTRO}
```
[the remark this answers is not in the document]
Comment (fzoric8) [reply ros2-humble/custom/Dockerfile:102]: No specific reason. We can go with binaries. @mkrizmancic correct it when adding gazebo. 

Pls try to add gazebo so I can try multistage build with one or both of those simulators :)

File: ros2-humble/custom/Dockerfile [hunk 2]
```diff
@@ -62,15 +60,49 @@ COPY ./to_copy/nanorc /root/.nanorc
...
+RUN echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
+
 # Init ROS2 workspace
-RUN mkdir -p ros2_ws/src
-WORKDIR /root/ros2_ws
+RUN mkdir -p /home/developer/ros2_ws/src
+WORKDIR /home/developer/ros2_ws
 RUN colcon build
-WORKDIR /root/ros2_ws/src
+WORKDIR /home/developer/ros2_ws/src
```
[the remark this answers is not in the document]
Comment (fzoric8) [reply ros2-humble/custom/Dockerfile:72]: I think it is related to this: https://forums.docker.com/t/using-ssh-to-access-private-data-in-builds-as-non-root-user/72881

However, I think that we can remove `developer` user if it complicates things.

**Comment by fzoric8** [comment]
[[Docker] Setup](https://trello.com/c/v4oxi6Zg/10-docker-setup)

**Review by mkrizmancic — APPROVED** [review]
Looks good. I tried a couple of demos from https://github.com/gazebosim/ros_gz/tree/ros2/ros_gz_sim_demos and there were no problems.

**Pull request merged**